### PR TITLE
Dev-2909 IDV Activity Chart API Integration

### PR DESCRIPTION
**Pull request by jonhill13**
**Do Not Merge**
- [x] https://github.com/fedspendingtransparency/usaspending-website/pull/1144 Loading messages
- [x] Considering Ross's comments in Slack `since this chart only shows contract award children/grandchildren of a given IDV, and not contract IDV children, we would expect ordering_period_end_date to be blank for everything shown here. if ordering_period_end_date is being returned as part of the API contract for this chart, or the basis for anything displayed on this chart, there seems to be a mistake and it needs to be remapped`. We are going to hold off on merging this PR until we figure out the correct mapping with design and the backend team.
https://federal-spending-transparency.atlassian.net/browse/DEV-2970

**High level description:**

Add API Call to IDV Activity Chart

**Technical details:**

* update api call
* add tests
* update mock api
* update mock data

**JIRA Ticket:**
[DEV-2909](https://federal-spending-transparency.atlassian.net/browse/DEV-2909)

The following are ALL required for the PR to be merged:
- [ ] Code review

### CODE REVIEW

**Comment by MaxwellKendall** [comment]
Perhaps this is a WIP still given the DNM label, just wanted to comment that I'm looking at this and it appears the response from the API routinely has a `null` value for the `last_date_to_order` which is not being anticipated in our models besides returning `--` in this case.


image

**Comment by jonhill13** [comment]
@MaxwellKendall Yeah, I saw that, and made a little adjustment to the model so it would not completely break the page. But the [ticket](https://federal-spending-transparency.atlassian.net/browse/DEV-2909) states `Let any bad data/ edge cases show. Those will be covered in DEV‌-2925, DEV‌-2926, DEV‌-2927`

**Comment by MaxwellKendall** [comment]
@jonhill13 can you provide an award id in dev that does not constitute an edge case?

**Comment by jonhill13** [comment]
@MaxwellKendall The short answer is no. I ran this query against the dev db with with the help of @emily-brents and no results returned.
`with gather_award_ids as (
        select  award_id,
                false grandchild
        from    parent_award
        union all
        select  cpa.award_id,
                true grandchild
        from    parent_award ppa
                inner join parent_award cpa on cpa.parent_award_id = ppa.award_id
    )
    select
        ca.id                                           award_id,
        ta.name                                         awarding_agency,
        ca.awarding_agency_id                           awarding_agency_id,
        ca.generated_unique_award_id,
        tf.ordering_period_end_date                     last_date_to_order,
        ca.total_obligation                             obligated_amount,
        ca.base_and_all_options_value                   awarded_amount,
        ca.period_of_performance_start_date,
        ca.piid,
        rl.legal_business_name                          recipient_name,
        rl.recipient_hash || case
            when tf.ultimate_parent_unique_ide is null then '-R'
            else '-C'
        end                                             recipient_id,
        gaids.grandchild
    from
        gather_award_ids gaids
        inner join awards pa on pa.id = gaids.award_id
        inner join awards ca on
            ca.parent_award_piid = pa.piid and
            ca.fpds_parent_agency_id = pa.fpds_agency_id and
            ca.type not like 'IDV%' 
        left outer join transaction_fpds tf on tf.transaction_id = ca.latest_transaction_id
        left outer join recipient_lookup rl on rl.duns = tf.awardee_or_recipient_uniqu
        left outer join agency a on a.id = ca.awarding_agency_id
        left outer join toptier_agency ta on ta.toptier_agency_id = a.toptier_agency_id
        where tf.ordering_period_end_date is not null
    order by
        ca.base_and_all_options_value desc, ca.id desc;`

**Comment by MaxwellKendall** [comment]
Appears now that `last_date_to_order` is returning not null and the chart is working.

File: tests/containers/awardV2/idv/IdvActivityContainer-test.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+describe('IdvActivityContainer', () => {
+    const loadAwards = jest.fn();
+    const parseAwards = jest.fn();
+    it('should make an API call for the awards on mount', async () => {
+        const container = mount(<IdvActivityContainer {...mockActions} {...mockRedux} />);
+
+        container.instance().loadAwards = loadAwards;
+        container.instance().parseAwards = parseAwards;
+        await container.instance().componentDidMount();
```
Comment (MaxwellKendall) [inline tests/containers/awardV2/idv/IdvActivityContainer-test.jsx]: are we awaiting a promise in this case? I don't think we typically return anything in the lifecycle methods. Just curious to get your thoughts on this given our previous convos.

**Pull request merged**